rust/src/model/sensors/sensor_type.rs

```diff
@@ -0,0 +1,77 @@
+use scylla::_macro_internal::CqlValue;
+use scylla::BufMut;
+use scylla::cql_to_rust::{FromCqlVal, FromCqlValError};
+use scylla::frame::value::{Value, ValueTooBig};
+use serde::Serialize;
+
+#[derive(Clone, Serialize)]
+pub enum SensorType {
+    #[serde(rename = "T")]
+    Temperature,
+    #[serde(rename = "P")]
+    Pulse,
+    #[serde(rename = "L")]
+    Location,
+    #[serde(rename = "R")]
+    Respiration,
+}
+
+impl FromCqlVal<CqlValue> for SensorType {
+    fn from_cql(cql_val: CqlValue) -> Result<Self, FromCqlValError> {
+        Ok(SensorType::from_str(cql_val
+            .as_text()
+            .ok_or(FromCqlValError::BadCqlType)?
+            .as_str()))
+    }
+}
+
+impl Value for SensorType {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        let bytes = self.as_str().as_bytes();
+        buf.put_i32(bytes.len().try_into().map_err(|_| ValueTooBig)?);
+        buf.put(bytes);
+
+        Ok(())
+    }
+}
+
+impl Default for SensorType {
```

Comment:
this can be derived if you add Default to SensorType, and #[default] to Temperature, then just get rid of this impl

```
#[derive(Clone, Serialize, Default)]
pub enum SensorType {
    #[serde(rename = "T")]
    #[default]
    Temperature,
    #[serde(rename = "P")]
    Pulse,
    #[serde(rename = "L")]
    Location,
    #[serde(rename = "R")]
    Respiration,
}
```